Website/src/rendering/src/components/Forms/AddressForm.tsx

```diff
@@ -0,0 +1,168 @@
+import { DBuyerAddress } from '../../models/ordercloud/DBuyerAddress';
+import { FormEvent, useState } from 'react';
+import { GeographyService } from '../../services/GeographyService';
+
+type AddressFormProps = {
+  address?: DBuyerAddress;
+  onSubmit?: (address: DBuyerAddress) => void;
+};
+
+const AddressForm = (props: AddressFormProps): JSX.Element => {
+  /**
+   * TODO:
+   * 1. Add better postal code validation based on country selected by using masks (react-input-mask)
+   * 2. Add more supported countries, currently only support US & Canada
+   * 3. Disable submit button unless there are actual changes
+   * 4. Add ability to discard changes (this would only be on editing addresses)
+   * 5. Remove mocked address once saved addresses are a thing
+   */
+  const countries = GeographyService.getCountries();
+  const [states, setStates] = useState(
+    GeographyService.getStatesOrProvinces(props.address?.Country || countries[0].code)
+  );
+  const [addressName, setAddressName] = useState(
+    props?.address ? props.address?.AddressName : 'Home'
+  );
+  const [country, setCountry] = useState(props?.address ? props.address?.Country : 'US');
+  const [street1, setStreet1] = useState(
+    props?.address?.ID ? props.address?.Street1 : '6818 Gaines Ferry Road'
+  );
+  const [street2, setStreet2] = useState(props.address?.Street2);
+  const [city, setCity] = useState(props?.address ? props.address?.City : 'Flowery Branch');
+  const [state, setState] = useState(props?.address ? props.address?.State : 'GA');
+  const [zip, setZip] = useState(props?.address ? props.address?.Zip : '30542');
+
+  const handleFormSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const updatedAddress = {
+      ...(props.address || {}),
+      AddressName: addressName,
+      Country: country,
+      Street1: street1,
+      Street2: street2,
+      City: city,
+      State: state,
+      Zip: zip,
+    };
+
+    if (props.onSubmit && typeof props.onSubmit === 'function') {
```

Comment:
Is the typeof function check necessary? The props already define onSubmit as a function.

Reply:
The props define it as an optional parameter. Its useful not having to define in storybook for example. If you prefer I can make it required and update our storybook examples though to do a no-op function.